client.go

```diff
@@ -11,14 +11,14 @@ import (
 // It implements `golang.org/x/net/websocket` under the hood and wraps it into a channel,
 // allowing it to be select-able along with other channels.
 type WebSocket struct {
-	Connected bool
-	Events    chan string
-	OnError   func(error)
-
-	url        string
-	client     *websocket.Conn
-	runCtx     context.Context
-	cancelFunc context.CancelFunc
+	Connected bool        // Connected indicates whether the WebSocket connection is currently active.
+	Events    chan string // Events is a channel for receiving WebSocket events and messages.
+	OnError   func(error) // OnError is a callback function that will be called in case of an error during WebSocket operation.
+
+	url       string             // url is the WebSocket server URL.
+	client    *websocket.Conn    // client is the underlying WebSocket connection.
+	context   context.Context    // runCtx is the context used for managing the WebSocket connection's lifecycle.
```

Comment:
The comment should be -> // context is the context used for managing the WebSocket connection's lifecycle.